### ISSUE TITLE
Consider removing the instance hashcode in the metric naming

### ISSUE THREAD

**Issue by vietj** [vert-x3/vertx-dropwizard-metrics#12]
For instance in HttpClient naming.

**Comment by leolux** [comment]
Will it become an aggregation per VM or per cluster?

**Comment by vietj** [comment]
I'm not sure to understand what you mean

**Comment by leolux** [comment]
Ok, I try to explain my questions. I am running vert.x in cluster mode whereby each node runs on a different machine. That means that each node has its own network address. Currently I fetch the metrics for this cluster seperately for each node using the appropiate address. My question is, if the instance hashcode will be removed will it change my process in the way, that I could fetch the metrics for all nodes at once by picking only a single node?

**Comment by vietj** [comment]
the instance hashcode is not related to Dropwizard collection on a cluster.

If you are going to collect in a cluster and get an aggregated view of the metrics, then you should configure Dropwizard on each node to collect the metrics in a shared store. Then the store would more or less provide a view of the cluster.

**Comment by leolux** [comment]
Would I be correct in saying that the standard dropwizard configuration does not use a shared store and as a consequence each node provides its own view of metrics aggregating only their own HttpClient handler instances?

**Comment by vietj** [comment]
yes, it is local to the Registry so not even shared between several instances in the same VM.

If you want shared metrics on a cluster, you should configure Dropwizard to report to a shared store: https://dropwizard.github.io/metrics/3.1.0/manual/ganglia/ , https://dropwizard.github.io/metrics/3.1.0/manual/graphite/ . In this situation you need to setup your own Registry configured this way in the SharedMetricsRegistries and then tell Vert.x Dropwizard Metrics to use it by specifying the name in the configuration options.

Alternatively you could use in the near future a different implementation of the Metrics SPI for Hawkular metrics (https://github.com/tsegismont/vertx-monitor) or InfluxDB (but this one is more a proof of concept https://github.com/vietj/vertx-influxdb-metrics) that are designed for this purpose.

**Comment by leolux** [comment]
Currently I don't need shared metrics for my cluster. Providing a seperate view of metrics for every node is totally fine for me at the moment. The main reason I don't like to switch to shared metrics is that I can leave my backend system unchanged which effectively means zero effort :-)

Thank you for your detailed explanation and the links. Likely I will have a look at Hawkular and InfluxDB in future.

**Comment by leolux** [comment]
"it is local to the Registry so not even shared between several instances in the same VM"
I guess you mean the registry it is not shared between several VM instances in the same runtime environment.

**Comment by marcust** [comment]
@vietj 
I'm not quite sure what this thread is about, but the fix broke multiple usage of HttpClient in 3.2.0.
Calling createMetrics multiple times with the same name throws an IllegalArgumentException out of the MetricsRegistry if there is one shared registry for the whole VM.

```
java.lang.IllegalArgumentException: A metric named vertx.http.clients.connections.max-pool-size already exists
	at com.codahale.metrics.MetricRegistry.register(MetricRegistry.java:91)
	at io.vertx.ext.dropwizard.impl.AbstractMetrics.gauge(AbstractMetrics.java:109)
	at io.vertx.ext.dropwizard.impl.HttpClientMetricsImpl.<init>(HttpClientMetricsImpl.java:40)
	at io.vertx.ext.dropwizard.impl.VertxMetricsImpl.createMetrics(VertxMetricsImpl.java:118)
	at io.vertx.core.http.impl.HttpClientImpl.<init>(HttpClientImpl.java:96)
	at io.vertx.core.impl.VertxImpl.createHttpClient(VertxImpl.java:238)
```

**Comment by cescoffier** [comment]
@marcust your issue has been fixed in https://github.com/vert-x3/vertx-dropwizard-metrics/issues/20.

**Comment by vietj** [comment]
yes it has been fixed and we should have a 3.2.1 release this month including this fix
